packages/dds/tree/src/feature-libraries/optional-field/optionalFieldCodecV0.ts

```diff
@@ -105,25 +106,28 @@ export function makeOptionalFieldCodec<TChildChange = NodeChangeset>(
 				encoded.c = [];
 				for (const [id, childChange] of change.childChanges) {
 					encoded.c.push([
-						registerIdCodec.encode(id, context),
-						childCodec.encode(childChange, context),
+						registerIdCodec.encode(id, context.baseContext),
+						context.encodeNode(childChange as NodeId),
```

Comment:
It would be nice to get rid of this cast here (and `context.decodeNode(encodedChange) as TChildChange` below). I think it would be easy enough to make `FieldChangeEncodingContext` take a template type to use instead of `NodeId`.

Reply:
I plan to remove the child type parameter from optional and sequence changesets as a follow up, as I thought doing so in this PR might increase the size of the diff significantly.

Reply:
Ah, right, so while adding a type param to FieldChangeEncodingContext would not be a big change, there's little incentive to do it since the removal of the child type parameter from optional and sequence will render it moot. Resolving conversation.